gym/utils/env_checker.py

```diff
@@ -8,477 +8,241 @@
-    """
-    # Helper to link to the code, because gym has no proper documentation
-    gym_spaces = " cf https://github.com/openai/gym/blob/master/gym/spaces/"
-
-    assert hasattr(env, "observation_space"), (
-        "You must specify an observation space (cf gym.spaces)" + gym_spaces
-    )
-    assert hasattr(env, "action_space"), (
-        "You must specify an action space (cf gym.spaces)" + gym_spaces
-    )
-
-    assert isinstance(env.observation_space, Space), (
-        "The observation space must inherit from gym.spaces" + gym_spaces
-    )
-    assert isinstance(env.action_space, Space), (
-        "The action space must inherit from gym.spaces" + gym_spaces
-    )
+        raise error.Error(
+            "The `reset` method does not provide the `options` keyword argument"
+        )
 
 
 # Check render cannot be covered by CI
-def _check_render(env: gym.Env, warn: bool = True, headless: bool = False):
+def check_render(env: gym.Env, headless: bool = False):
     """Check the declared render modes/fps and the :meth:`render`/:meth:`close` method of the environment.
 
     Args:
         env: The environment to check
-        warn: Whether to output additional warnings
         headless: Whether to disable render modes that require a graphical interface. False by default.
     """
     render_modes = env.metadata.get("render_modes")
     if render_modes is None:
-        if warn:
-            logger.warn(
-                "No render modes was declared in the environment "
-                " (env.metadata['render_modes'] is None or not defined), "
-                "you may have trouble when calling `.render()`"
-            )
+        logger.warn(
+            "No render modes was declared in the environment  (env.metadata['render_modes'] is None or not defined), you may have trouble when calling `.render()`"
+        )
 
     render_fps = env.metadata.get("render_fps")
     # We only require `render_fps` if rendering is actually implemented
-    if render_fps is None and render_modes is not None and len(render_modes) > 0:
-        if warn:
-            logger.warn(
-                "No render fps was declared in the environment "
-                " (env.metadata['render_fps'] is None or not defined), "
-                "rendering may occur at inconsistent fps"
-            )
+    if render_fps is None:
+        logger.warn(
+            "No render fps was declared in the environment (env.metadata['render_fps'] is None or not defined), rendering may occur at inconsistent fps"
+        )
 
-    else:
+    if render_modes is not None:
         # Don't check render mode that require a
         # graphical interface (useful for CI)
         if headless and "human" in render_modes:
             render_modes.remove("human")
+
         # Check all declared render modes
-        for render_mode in render_modes:
-            env.render(mode=render_mode)
+        for mode in render_modes:
+            env.render(mode=mode)
```

Comment:
This will not work anymore with the new Render API.

Initially, I thought to just change the env_checker to receive the environment specification. In this way, internally it can creates as many copies as it needs without touching the copy made for the user. Having the env_spec here will also solve the problem for rendering (and in general for checking the parameters that are passed to`__init__`)

Reply:
I don't total get what you mean, but as this PR is going to be in the next sub-release, would you be able to make the changes in your render PR?